### PR TITLE
Reaction handling

### PR DESCRIPTION
**Pull request by loredous**
Added handling of Reaction messages. Provides text value of reaction, user that posted the reaction, and timestamp of the item being reacted to.

### CODE REVIEW

File: src/SlackConnector/Connections/Sockets/Messages/Inbound/MessageInterpreter.cs [hunk 1]
```diff
@@ -84,5 +87,20 @@ private static ChatMessage GetChatMessage(string json)
...
+        private static ReactionMessage GetReactionMessage(string json)
+        {
+            var message = JsonConvert.DeserializeObject<ReactionMessage>(json);
+
+            if (message != null)
+            {
+                JObject messageJobject = JObject.Parse(json);
+                message.Channel = messageJobject["item"]["channel"].Value<string>();
+                message.ReactingToTimestamp = messageJobject["item"]["ts"].Value<double>();
+                message.RawData = json;
```
Comment (Workshop2) [inline src/SlackConnector/Connections/Sockets/Messages/Inbound/MessageInterpreter.cs]: This is handled in the method that calls this

File: src/SlackConnector/Connections/Sockets/Messages/Inbound/MessageInterpreter.cs [hunk 2]
```diff
@@ -84,5 +87,20 @@ private static ChatMessage GetChatMessage(string json)
 
             return message;
         }
+
+        private static ReactionMessage GetReactionMessage(string json)
+        {
+            var message = JsonConvert.DeserializeObject<ReactionMessage>(json);
```
Comment (Workshop2) [inline src/SlackConnector/Connections/Sockets/Messages/Inbound/MessageInterpreter.cs:94]: Why are you parsing this object and then overriding the properties later? Can't this just be done in the de-serialise?

File: src/SlackConnector/Connections/Sockets/Messages/Inbound/MessageInterpreter.cs [hunk 3]
```diff
@@ -42,6 +42,9 @@ public InboundMessage InterpretMessage(string json)
                     case MessageType.Pong:
                         message = JsonConvert.DeserializeObject<PongMessage>(json);
                         break;
+                    case MessageType.Reaction_Added:
```
Comment (Workshop2) [inline src/SlackConnector/Connections/Sockets/Messages/Inbound/MessageInterpreter.cs:46]: Missing tests

File: src/SlackConnector/SlackConnection.cs [hunk 4]
```diff
@@ -95,6 +95,7 @@ private Task ListenTo(InboundMessage inboundMessage)
                 case MessageType.Im_Created: return HandleDmJoined((DmChannelJoinedMessage)inboundMessage);
                 case MessageType.Team_Join: return HandleUserJoined((UserJoinedMessage)inboundMessage);
                 case MessageType.Pong: return HandlePong((PongMessage)inboundMessage);
+                case MessageType.Reaction_Added: return HandleReaction((ReactionMessage)inboundMessage);
```
Comment (Workshop2) [inline src/SlackConnector/SlackConnection.cs:99]: Missing tests

**Review by Workshop2 — CHANGES_REQUESTED** [review]
Thanks for helping out. You are missing unit tests against the new logic and see my comments about the JSON parsing.

**Comment by loredous** [comment]
Added requested changes plus a few enhancements to deal with all current Reaction message types. Added unit tests for all new functionality.

**Pull request closed without merging**